ipatests/test_integration/test_insecure_crypto_SCEP.py

```diff
@@ -0,0 +1,141 @@
+#
+
+import glob
+
+from ipaplatform.paths import paths
+
+from ipatests.pytest_ipa.integration import tasks
+from ipatests.test_integration.base import IntegrationTest
+
+AUTHORIZATION_FILE = '/var/lib/pki/pki-tomcat/ca/conf/flatfile.txt'
+CERTMONGER_CONFIG = '/etc/sysconfig/certmonger'
+
+
+def enable_SCEP(host):
+    """Enable SCEP in Certificate Authority's CS.cfg
+    by setting ca.scep.enable=true
+    """
+    content = host.get_file_contents(paths.CA_CS_CFG_PATH,
+                                     encoding='utf-8')
+    new_lines = []
+    input_line = "auths.instance.flatFileAuth.deferOnFailure=false"
+    for line in content.split('\n'):
+        if line.startswith('auths.instance.flatFileAuth.deferOnFailure'):
+            new_lines.append(input_line)
+        elif line.startswith('ca.scep.enable'):
+            new_lines.append("ca.scep.enable=true")
+        else:
+            new_lines.append(line)
+    host.put_file_contents(paths.CA_CS_CFG_PATH, '\n'.join(new_lines))
+
+
+def add_local_ip_to_authfile(host):
+    """Add the local IP to the authorization file,
+    /var/lib/pki/pki-tomcat/ca/conf/flatfile.txt
+    """
+    content = host.get_file_contents(AUTHORIZATION_FILE,
+                                     encoding='utf-8')
+    new_lines = []
+    for line in content.split('\n'):
+        if line.startswith('#UID:'):
+            new_lines.append("UID:%s" % host.ip)
+        elif line.startswith("#PWD:"):
+            new_lines.append("PWD:1234")
+        else:
+            new_lines.append(line)
+    host.put_file_contents(AUTHORIZATION_FILE, '\n'.join(new_lines))
+
+
+def edit_certmonger_config(host):
+    """Increase certmonger debug output by setting
+    OPTS=-d 3 in /etc/sysconfig/certmonger
+    """
+    content = host.get_file_contents(CERTMONGER_CONFIG,
+                                     encoding='utf-8')
+    new_lines = []
+    for line in content.split('\n'):
+        if line.startswith('OPTS='):
+            new_lines.append("OPTS=-d 3")
+        else:
+            new_lines.append(line)
+    host.put_file_contents(CERTMONGER_CONFIG, '\n'.join(new_lines))
+
+
+def set_scep_cipher(host):
+    """hard code the default SCEP cipher to DES3 since there
+    is no CLI to do this by adding scep_cipher=DES3
+    """
+    input_file = None
+    for file in glob.glob('/var/lib/certmonger/cas/*'):
+        with open(file) as lines:
+            for line in lines:
+                if 'scep' in line:
+                    input_file = file
+    content = host.get_file_contents(input_file, encoding='utf-8')
+    content = '\n'.join([content, 'scep_cipher=DES3'])
+    host.put_file_contents(input_file, content)
+
+
+class TestInsecureCryptoSCEP(IntegrationTest):
+    """This test checks that certmonger does not support
+    insecure cryptography for SCEP enrollment
+    """
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
```

Comment:
You don't need the install method of you add ``topology = 'star'`` class attribute.

Reply:
added the topology